apps/server/src/modules/task/uc/task-copy.uc.ts

```diff
@@ -25,45 +20,72 @@ export class TaskCopyUC {
 
 	async copyTask(userId: EntityId, taskId: EntityId, parentParams: TaskCopyParentParams): Promise<CopyStatus> {
 		this.featureEnabled();
-		const user = await this.authorisation.getUserWithPermissions(userId);
-		const originalTask = await this.taskRepo.findById(taskId);
-		if (!this.authorisation.hasPermission(user, originalTask, AuthorizationContextBuilder.read([]))) {
-			throw new NotFoundException('could not find task to copy');
-		}
 
-		const destinationCourse = await this.getDestinationCourse(parentParams.courseId);
-		if (parentParams.courseId) {
-			await this.authorisation.checkPermissionByReferences(
-				userId,
-				AuthorizableReferenceType.Course,
-				parentParams.courseId,
-				{
-					action: Action.write,
-					requiredPermissions: [],
-				}
-			);
+		// i put it to promise all, it do not look like any more information can be expose over errors if it is called between the authorizations
+		// TODO: Add try catch around it with throw BadRequest invalid data
```

Comment:
Todo: speak with someone that develop the base code.

Reply:
Musst du hier nochmal mit jemandem reden? Es sind auch noch einige Kommentare hier im Code, die sich nach Klärungsbedarf anhören.

Reply:
Ja das wäre hilfreich gewesen, wenn sich jemand gefunden hätte. :D